### PR TITLE
Add reading PE export symbols to PEFile

### PR DESCRIPTION
**Pull request by mikem8361**

### CODE REVIEW

File: src/Microsoft.FileFormats/PE/PEFile.cs [hunk 1]
```diff
@@ -105,6 +106,55 @@ public bool IsValid()
...
+            }
+            offset = 0;
+            return false;
+        }
+
+        private ImageExportDirectory ReadExportDirectory()
+        {
+            if (IsValid())
+            {
+                ImageDataDirectory exportTableDirectory = ImageDataDirectory[(int)ImageDirectoryEntry.Export];
```
Comment (hoyosjs) [inline src/Microsoft.FileFormats/PE/PEFile.cs:149]: Is this resilient to only this section being available? I think PEFile expects to see all of the ImageDataDirectory, which is not even true in the PE spec for a regular image. (It assumes that always 15 sections will be available, completely ignoring `SizeOfHeaders`)

Comment (hoyosjs) [reply src/Microsoft.FileFormats/PE/PEFile.cs:149]: Can we assert that the Characteristics == 0?

File: src/Microsoft.FileFormats/PE/PEFile.cs [hunk 2]
```diff
@@ -105,6 +106,55 @@ public bool IsValid()
...
+                    for (int nameIndex = 0; nameIndex < exportDirectory.NumberOfNames; nameIndex++)
+                    {
+                        uint namePointerRVA = RelativeVirtualAddressReader.Read<uint>((ulong)(exportDirectory.AddressOfNames + (sizeof(uint) * nameIndex)));
+                        if (namePointerRVA != 0)
+                        {
+                            string name = RelativeVirtualAddressReader.Read<string>(namePointerRVA);
+                            if (name == symbolName)
+                            {
+                                ushort ordinalForNamedExport = RelativeVirtualAddressReader.Read<ushort>((ulong)(exportDirectory.AddressOfNameOrdinals + (sizeof(ushort) * nameIndex)));
+                                offset = RelativeVirtualAddressReader.Read<uint>((ulong)(exportDirectory.AddressOfFunctions + (sizeof(uint) * ordinalForNamedExport)));
```
Comment (hoyosjs) [inline src/Microsoft.FileFormats/PE/PEFile.cs:131]: This does not consider the case of a forward RVA export, thought that is an edge case that might not apply to our case. If `offset` is within the export section, then according to https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#export-address-table it's a pointer to a string that forwards the export to another dll potentially.

Comment (hoyosjs) [reply src/Microsoft.FileFormats/PE/PEFile.cs:131]: Can we check `ordinalForNamedExport` < `NumberOfFunctions`?

**Review by hoyosjs — APPROVED** [review]

File: src/Microsoft.FileFormats/PE/PEFile.cs (hunk 1 above)
Comment (mikem8361) [reply src/Microsoft.FileFormats/PE/PEFile.cs:149]: The symstore PE reader has always assumed that there are 15 sections and that they all are in the dump. It reads all 15 in ReadImageDataDirectory().  It sounds like you think it is important enough to only read what fits in SizeOfHeaders?  Does SizeOfHeaders include the size of the file header (ImageFileHeader) and optional header (ImageOptionalHeader)?

Should it throw an exception or do Debug.Assert if Characteristics != 0? A debug symstore isn't ever really used.

File: src/Microsoft.FileFormats/PE/PEFile.cs (hunk 2 above)
Comment (mikem8361) [reply src/Microsoft.FileFormats/PE/PEFile.cs:131]: The runtime PEDecoder doesn't check for either a forward RVA export or if `ordinalForNamedExport < NumberOfFunctions`.

File: src/Microsoft.FileFormats/PE/PEFile.cs (hunk 1 above)
Comment (hoyosjs) [reply src/Microsoft.FileFormats/PE/PEFile.cs:149]: Not sure if throw bad image, or say I can't read (return null);

**Pull request merged**